raster-foundry/raster-foundry
# Add organization (users/teams) and Team (users) crud views/functions

## Overview

Add list views for users on Organizations and Teams
Add List view for Teams on Organization
Hook up user avatar lists in user list views
Implement "add users" modal
Allow adding users to teams
Hook up search bars, pagination, and basic loading icon/animation

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~
~- [ ] Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~

### Demo
![image](https://user-images.githubusercontent.com/4392704/40143269-00591750-5929-11e8-8ab6-36c11be709c2.png)
![image](https://user-images.githubusercontent.com/4392704/40143280-063ebfa8-5929-11e8-8cbb-5b6817637fa5.png)
![image](https://user-images.githubusercontent.com/4392704/40143291-0d1f797a-5929-11e8-987c-2199326c8f60.png)
![image](https://user-images.githubusercontent.com/4392704/40143303-143a64cc-5929-11e8-9816-2a65f9fd512d.png)
![image](https://user-images.githubusercontent.com/4392704/40143311-1a30865e-5929-11e8-938b-23b4386349c1.png)
![image](https://user-images.githubusercontent.com/4392704/40143335-255235e6-5929-11e8-88a0-82db0b23d6dc.png)
![image](https://user-images.githubusercontent.com/4392704/40143346-2afd50d4-5929-11e8-8255-624198487b1d.png)

### Notes
This has not been rebased to develop yet, so the search bars make the correct requests, but the backend does not filter.

### Todo
- [ ] Add organization name to organization page
- [ ] Add platform name to platform page
- [ ] Rebase once #3381 is merged so searches can be tested correctly

## Testing Instructions
* Give several users admin / member roles on a platform and organization
* Verify that the organization user list page works
* Verify that the organization team page works, and shows user icons correctly
* Verify that the team user list page works
* Verify that search bars in the admin views make the correct requests when text is entered into them
* Verify that users can be added to teams
* Verify that pagination works on these views: platform organization list, platform user list, organization user list, organization team list, team user list, team add user modal

Closes https://github.com/raster-foundry/raster-foundry/issues/3361
Closes https://github.com/raster-foundry/raster-foundry/issues/3249
Closes https://github.com/raster-foundry/raster-foundry/issues/3248
Depends on https://github.com/raster-foundry/raster-foundry/pull/3381
